### ISSUE TITLE
Maps with Google background sometimes fail when openlayers is selected

### ISSUE THREAD

**Issue by offtherailz** [geosolutions-it/MapStore2#2095]
#### Steps to reproduce: 
 - Open home page
 - Select OpenLayers as maptype
 - Open the map 749 (named MS2 GTOPO30) 
 - Back to the home page clicking on the home button
 - Open 1740 (WFS Search) 

 #### Expected result:
 The map should open

#### Effective result: 
The page shows only "Loading...". Click on home and select again the map will solve the problem. 

#### Details
From the debugger i notice the application tries to invoke Google Maps API 
```
gmaps[mapId] = new google.maps.Map(document.getElementById(mapId + 'gmaps'), {
                disableDefaultUI: true,
                keyboardShortcuts: false,
                draggable: false,
                disableDoubleClickZoom: true,
                scrollwheel: false,
                streetViewControl: false
            });
```
But the  `document.getElementById(mapId + 'gmaps'),` element doesn't exist (anymore). 

So this Error is thrown: 
```
js?v=3:87 Uncaught TypeError: Cannot read property 'firstChild' of null
    at Object._.vg (js?v=3:87)
    at new Ag (js?v=3:89)
    at eval (eval at create (GoogleLayer.js:NaN), <anonymous>:1:16)
    at Object.create (GoogleLayer.js:35)
    at Object.createLayer (Layers.js:22)
    at OpenlayersLayer._this.createLayer (Layer.jsx:96)
    at OpenlayersLayer.componentDidMount (Layer.jsx:182)
    at OpenlayersLayer.eval (makeAssimilatePrototype.js:15)
    at eval (ReactCompositeComponent.js:265)
    at measureLifeCyclePerf (ReactCompositeComponent.js:75)

```
You will notice also this error in console. 
Error in console: 
```
Uncaught TypeError: Cannot read property 'id' of undefined
    at eval (map.js:43)
    at FilterSubscriber._next (filter.js:81)
    at FilterSubscriber.Subscriber.next (Subscriber.js:89)
    at Notification.observe (Notification.js:32)
    at AsyncAction.DelaySubscriber.dispatch (delay.js:88)
    at AsyncAction._execute (AsyncAction.js:111)
    at AsyncAction.execute (AsyncAction.js:86)
    at AsyncScheduler.flush (AsyncScheduler.js:36)
```

#### Possible reason
The `document.getElementById(mapId + 'gmaps')` element should be created in the render method. Probably render cycles mixed between old and new map do not allow to render before to invoke GoogleMaps API.

**Comment by mbarto** [comment]
Can you still reproduce this one?

**Comment by offtherailz** [comment]
Yes I can. Switching from maps with Google background and without it that background. 
This is a partial fix of the issue, You may need some unit tests.
[patch_2095.txt](https://github.com/geosolutions-it/MapStore2/files/1568778/patch_2095.txt)

**Comment by mbarto** [comment]
@offtherailz can you give two maps corresponding to your description?

**Comment by tdipisa** [comment]
Fixed in 2018.02.00
